.github/workflows/build-binaries-and-push-to-r2.yaml

```diff
@@ -0,0 +1,46 @@
+name: Upload core binary to Cloudflare R2 bucket
+
+on:
+  push:
+    branches:
+      - '*'
```

Comment:
Do we want to run it on  push on every branch  ?

Reply:
Yeah. Based on our talks with @talalashraf, in another repository, we want people to be able to input any commit hash and then retrieve the binary of that commit. But we don't want to build binaries on that repository so we put the action here.